Sources/Plasma/NucleusLib/pnNetProtocol/Private/pnNpCommon.cpp

```diff
@@ -499,9 +499,9 @@ inline void IRead<ST::string>(const uint8_t*& buf, ST::string& dest)
     buf += sizeof(uint32_t);
 
     ST::utf16_buffer str;
-    char16_t* theStrBuffer = str.create_writable_buffer(nChars);
-    memcpy(theStrBuffer, buf, size);
-    theStrBuffer[nChars] = 0;
+    str.allocate(nChars);
+    memcpy(str.data(), buf, size);
+    str[nChars] = 0;
```

Comment:
This is playing with my mind a little bit since we're juggling both the buffer size and the string length. Perhaps it would be more approachable if we only memcpy'd nChars and removed line 504 altogether?

Reply:
Agreed, and it would match the other updates better too.
+Fixed